ember-core/shared/src/main/scala/org/http4s/ember/core/Encoder.scala

```diff
@@ -63,8 +64,12 @@ private[ember] object Encoder {
           ()
         }
       }
-      if (!chunked && !appliedContentLength && resp.status.isEntityAllowed) {
-        stringBuilder.append(chunkedTansferEncodingHeaderRaw).append(CRLF)
+      if (!appliedContentLength && resp.body == EmptyBody && resp.status.isEntityAllowed) {
```

Comment:
The logic here looks correct, but it's observable behavior that's dependent on reference equality of a stream.  They're both "correct" HTTP, and this is a better outcome for a common case.  I think I'm okay with this, but calling it out.

In main, with the new entity model, we can target the Empty entity and not have this concern.

Reply:
I looked at it, and I could not see any nice way to check whether a fs2 stream is empty. Maybe somebody can enlighten me.

Reply:
I am still confused why it was broken with an empty body and the chunked encoding.

Reply:
There really isn't a good way.  We can "peek" a stream, but that semantically blocks until the first chunk of the body (if any) is produced, and would delay rendering the headers.  In scalaz-stream, we used to be able to introspect the algebra and find a wider class of empty streams, but still not all off them.  Effects are, well, effectful.

Reply:
Ah ok. Makes sense :+1:

Reply:
> I think I'm okay with this, but calling it out.

Seems fine to me.